### ISSUE TITLE
Cannot find -lfftw3 with SPEC_ROOT on Ocean

### ISSUE THREAD

**Issue by mar-celine** [sxs-collaboration/spectre#5271]
# Bug reports:

Expecting -lfftw3 to be found when `SPEC_ROOT` is set.
Currently linking only succeeds with `SPEC_ROOT` is unset.
@nilsvu says this is related to @nilsdeppe 's #5250 

This is on Ocean with gcc.

### Current behavior:
When building,
`/usr/bin/ld.gold: error: cannot find -lfftw3`

### Environment:

GSL version: 2.7.1
HDF5 version: 1.12.2
jemalloc version: 5.2.1
SpEC exporter: /opt/ohpc/pub/apps/spec/spec-interp4/Support/ApplyObservers/Exporter
Catch version: 2.13.8
NumPy version: 1.21.1
CMake version: 3.24.1
CMAKE_CXX_FLAGS:
CMAKE_CXX_LINK_FLAGS:  -fuse-ld=gold
CMAKE_CXX_FLAGS_DEBUG: -g -DSPECTRE_DEBUG
CMAKE_CXX_FLAGS_RELEASE: -O3 -DNDEBUG
CMAKE_C_FLAGS:
CMAKE_C_FLAGS_DEBUG: -g
CMAKE_C_FLAGS_RELEASE: -O3 -DNDEBUG
CMAKE_Fortran_FLAGS:
CMAKE_Fortran_FLAGS_DEBUG: -g
CMAKE_Fortran_FLAGS_RELEASE: -O3 -DNDEBUG -O3
CMAKE_EXE_LINKER_FLAGS:
CMAKE_BUILD_TYPE: Release
CMAKE_CXX_COMPILER: /opt/ohpc/pub/compiler/gcc/11.3.0/bin/g++
CMAKE_CXX_COMPILER_VERSION: 11.3.0
CMAKE_C_COMPILER: /opt/ohpc/pub/compiler/gcc/11.3.0/bin/gcc
CMAKE_C_COMPILER_VERSION: 11.3.0
CMAKE_Fortran_COMPILER: /opt/ohpc/pub/compiler/gcc/11.3.0/bin/gfortran
CMAKE_Fortran_COMPILER_VERSION: 11.3.0
Python version: 3.9.5

**Comment by nilsvu** [comment]
I think the problem is that `FFTW_HOME` is not expanded when parsing the machine file from spec.

**Comment by nilsdeppe** [comment]
Yes, sort of. I'm not sure which of the many ocean files is used, but this looks like the FFTW module is not configured correctly. This is an upstream issue with Ocean's configuration, not with how we parse the line. Both SpEC's and SpECTRE's config on ocean should load the FFTW module, not set paths explicitly. 

Edit: I'm not sure this ever worked on Ocean, and if it did then I really don't know how because the current `Ocean_gcc.def` in SpEC loads FFTW dynamically, which SpECTRE couldn't handle until my change. The lines:
```shell
FFTW_INCLUDE = -I$(FFTW_HOME)/include
FFTW_LIB     = -Wl,-rpath,$(FFTW_HOME)/lib -L$(FFTW_HOME)/lib -lfftw3
```
shouldn't need anything other than
```shell
FFTW_INCLUDE =
FFTW_LIB = -lfftw3
```
Anything else is a module configuration problem on Ocean, which should be fixable since @geoffrey4444 has root. The Ocean modules need to append:
```shell
CPATH=$(FFTW_HOME)/include
LD_LIBRARY_PATH=$(FFTW_HOME)/lib
LIBRARY_PATH=$(FFTW_HOME)/lib
```

**Comment by nilsvu** [comment]
Maybe on the spectre side we should find and link FFTW3 like we find and link MPI (`find_package(fftw3)` and `target_link_libraries(SpEC::Exporter INTERFACE fftw3`), but only if `-lfftw3` is found in the spec machine file (meaning we only find and link fftw3 if spec links it dynamically. Else we don't have to do anything).

Edit: Then we don't have to rely on the specifics of the SpEC machine file.

**Comment by nilsdeppe** [comment]
Yea, that would probably be safer. I think the issue here would still persist here since it's a module configuration problem, not how we find FFTW

**Comment by geoffrey4444** [comment]
I will fix when back from India

**Comment by geoffrey4444** [comment]
I tried to fix this, but I still get the error, even after confirming that my fftw3 module does in fact set CPATH, LD_LIBRARY_PATH, and LIBRARY_PATH as suggested above, and even after changing my spec's this_machine.def to only put in `-lfftw3`. I'm not sure why, when my fftw-3 module is loaded, I'm still getting this error.

I wonder if the cmake change suggested above would fix this, but I won't be able to try this any sooner than Wednesday (out of the office tomorrow).

Before, I got no error, because when I built the exporter library on ocean I modified `Support/ApplyObservers/Exporter/Makefile` to read

```
ExporterInhibitFactoryRegistration.o /opt/ohpc/pub/apps/spack2022/opt/spack/linux-centos7-broadwell/gcc-11.3.0/fftw-3.3.10-g2odatgw5xazimxconntsxknzwdi5pxe/lib/libfftw3.a $$Deps;
```

instead of

```
ExporterInhibitFactoryRegistration.o $(FFTW_LIB) $$Deps;
```
